subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheValueSourceIntegrationTest.groovy

```diff
@@ -292,4 +295,76 @@ class ConfigurationCacheValueSourceIntegrationTest extends AbstractConfiguration
             ignoringUnexpectedInputs()
         }
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/22305")
+    def "evaluating a value source parameter does not enable input tracking inside the value source"() {
+        given:
+        def configurationCache = newConfigurationCacheFixture()
+        buildFile.text = """
+
+            import org.gradle.api.provider.*
+
+            abstract class MySource implements ValueSource<String, Parameters> {
+                interface Parameters extends ValueSourceParameters {
+                    Property<String> getValue()
+                }
+                @Override String obtain() {
+                    def suffix = parameters.value.orElse("").get()
+                    return System.getProperty("my.property") + suffix
+                }
+            }
+
+            def someProp = providers.systemProperty("some.property")
+
+            def vsResult = providers.of(MySource) {
+                parameters.value = someProp
+            }
+
+            println("ValueSource result = \${vsResult.get()}")
+        """
+
+        when:
+        configurationCacheRun("-Dsome.property=1", "-Dmy.property=value")
+
+        then:
+        configurationCache.assertStateStored()
+        outputContains("result = value1")
+        problems.assertResultHasProblems(result) {
+            withInput("Build file 'build.gradle': system property 'some.property'")
```

Comment:
❓ I assume this will no longer be the case (`some.property` being an input) once https://github.com/gradle/gradle/issues/22305 is fixed?

Reply:
No, we'll need #22494 for that. This PR fixes #22305 which is essentially "all inputs are now tracked in my `ValueSource.obtain()` implementation after I called `parameters.myParam.get()`".  Without the fix, we'd also have `system property my.property` as an input.

Reply:
Got it, thanks.